internal/storage/fake/bucket.go

```diff
@@ -465,14 +467,45 @@ func (b *bucket) ListObjects(
 	// Find the range of indexes within the array to scan.
 	indexStart := b.objects.lowerBound(nameStart)
 	prefixLimit := b.objects.prefixUpperBound(req.Prefix)
-	indexLimit := minInt(indexStart+maxResults, prefixLimit)
+	var indexLimit int
+	checkForUnsupportedObjectNames := true
+	if maxResults == 1 {
+		// If maxResults is set to 1 (generally for LookUpInode), do not let
```

Comment:
> there won't be any error with MaxResults 1 even if the object is unsupported

This is obvious from the fact that the customer issues (ones having gs://bucket//a/b/c type objects) related to this have never had any mount issues, only listing issues.

> can we add a test for this case

I agree that we could add tests for this. I'd recommend adding a integration test for this.

> It is impossible from GCSFuse POV to get LookUpInode calls for unsupported file names.

It should be impossible because lookups of files/dirs of type a//b are always be resolved to a/b by linux kernel. Same goes for a/./b, a/../b and other types.

> If the above understanding is correct, I think it is okay to keep MaxResults parameter tests out of scope of this PR , and we can use maxResult 5000 for all tests. ( I think it is okay because otherwise our test implementation is varying significantly from actual implementation).

@ashmeenkaur  Let's discuss this offline as it's an involved topic, plus I'm not sure I fully understand what you're saying.

Reply:
Obsolete comment. Logic to filter out unsupported objects has been moved up to the dirInode.readObjects() method.